internal/collector/otel_collector_plugin.go

```diff
@@ -117,19 +119,182 @@ func (oc *Collector) Close(ctx context.Context) error {
 	slog.InfoContext(ctx, "Closing OTel Collector plugin")
 
 	if !oc.stopped {
+		slog.InfoContext(ctx, "Shutting down OTel Collector", "state", oc.service.GetState())
 		oc.stopped = true
```

Comment:
Just in the interest of correctness, should we toggle this only when state is `otelcol.StateClosed` in our backoff function?